### PR TITLE
poc: dataclasses for readonly fields and commitment, document support

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: sage_desktop_sdk/core/schema/read_only.py [hunk 1]
```diff
@@ -2,15 +2,9 @@
...
-    Name: int
-    ParentCode: str
-    ParentId: str
-    ParentName: str
+    id: str
+    code: str
+    version: int
+    is_active: bool
+    is_archived: bool
+    name: int
```
Comment (Shwetabhk) [inline sage_desktop_sdk/core/schema/read_only.py:10]: add a class function here
```python
@classmethod
    def from_dict(cls, account_dict):
        return cls(
            id=account_dict['Id'],
            # ... other fields ...
        )
```

File: sage_desktop_sdk/apis/accounts.py [hunk 2]
```diff
@@ -2,16 +2,31 @@
...
+from sage_desktop_sdk.core.schema.read_only import Account
 
 
 class Accounts(Client):
     """Class for Accounts APIs."""
 
     GET_ACCOUNTS = '/GeneralLedger/Api/V1/Account.svc/accounts'
 
+    @classmethod
+    def generate_accounts(cls, account_data):
```
Comment (Shwetabhk) [inline sage_desktop_sdk/apis/accounts.py:14]: remove this method

File: sage_desktop_sdk/apis/accounts.py [hunk 3]
```diff
@@ -2,16 +2,31 @@
...
+
     def get_all(self):
         """
         Get all Attachables
         :return: List of Dicts in Attachable Schema
         """
-        return self._query_get_all(Accounts.GET_ACCOUNTS)
+
+        accounts = self._query_get_all(Accounts.GET_ACCOUNTS)        
+        return self.generate_accounts(accounts)
```
Comment (Shwetabhk) [inline sage_desktop_sdk/apis/accounts.py:32]:
```suggestion
        for account in accounts:
            yield Account.from_dict(account)
```

File: sage_desktop_sdk/core/schema/read_only.py (hunk 1 above)
Comment (NileshPant1999) [reply sage_desktop_sdk/core/schema/read_only.py:10]: @Shwetabhk we are not using yield here ??

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**